### PR TITLE
Refactor dependencies and introduce sonarqube instead of coveralls

### PR DESCRIPTION
**Pull request by rosolko**
## Proposed changes
Linked with #702 
Refactor dependencies and introduce sonarqube instead of coveralls.
For now this PR will fail because it's required at least sonar organization and login key.
But as said documentation sonarqube task from travis script can be replaced with plugins, but I didn't test plugin method.
```
addons:
  sonarcloud:
    organization: "your_organization_key"
    token:
      secure: "your_token_encrypted_with_travis_encrypt_tool"
```
Also need to configure readme file badges with sonar links.
Local sonar instance: https://sonarcloud.io/dashboard?id=com.codeborne%3Aselenide

Also I introduce automatic module names to make at least first step into java 9/10.

Run test tasks to proof compatibility: `check`, `chrome`, `htmlunit` and `grid`

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,8 +26,11 @@ before_install:
...
  - rm -fr /home/travis/.m2/repository/webdriver/
 
-after_success:
-- ./gradlew jacocoTestReport coveralls
-- bash <(curl -s https://codecov.io/bash)
+script:
+  - ./gradlew check chrome htmlunit --no-daemon --console=plain
+  - ./gradlew sonarqube --no-daemon --console=plain -Dsonar.organization=${SONAR_ORG} -Dsonar.host.url=${SONAR_HOST} -Dsonar.login=${SONAR_LOGIN} -Dsonar.branch.name=${TRAVIS_BRANCH}
 
-after_script: "./.upload_reports.sh"
```
Comment (BorisOsipov) [inline .travis.yml]: is it ok? does sonar show failed tests stacktrace and etc?

File: .travis.yml [hunk 2]
```diff
@@ -1,23 +1,21 @@
-branches:
-  only:
-    - master
```
Comment (BorisOsipov) [inline .travis.yml]: why do you remove it?

Comment (BorisOsipov) [reply .travis.yml]: I mean does it correct to report to sonarqube from any forks\branches?

File: gradle/tests.gradle [hunk 3]
```diff
@@ -70,16 +70,18 @@ task testng(type: Test) {
...
-tasks.withType(Test).all { testTask ->
-  testTask.systemProperties['file.encoding'] = 'UTF-8'
-  testTask.systemProperties['BUILD_URL'] = System.getenv()['BUILD_URL']
-  // testTask.maxParallelForks = 2
-  testTask.systemProperties += System.properties
-  testTask.jacoco {
+tasks.withType(Test) {
+  systemProperty('file.encoding', encoding)
+  systemProperty('BUILD_URL', System.getenv()['BUILD_URL'])
+//  maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
```
Comment (BorisOsipov) [inline gradle/tests.gradle]: commented code

File: .travis.yml (hunk 2 above)
Comment (rosolko) [reply .travis.yml]: It's add possibility to build and perform code analysis for all branches.
And as I understood it necessary to process PRs.

File: gradle/tests.gradle (hunk 3 above)
Comment (rosolko) [reply gradle/tests.gradle]: Was commented before changes. 
I'm only update script that calculate available processors.
Based on gradle performance guide: https://guides.gradle.org/performance/#running_tests

File: .travis.yml (hunk 1 above)
Comment (rosolko) [reply .travis.yml]: Sonar isn't.
Failed tests stacktraces will on travis side.

Comment (BorisOsipov) [reply .travis.yml]: `"./.upload_reports.sh"` uses to upload test artifacts (in fact /build/reports/ folder) for test failure investigations. Please come this back

**Comment by rosolko** [comment]
Close due to not implements [feature for PRs](https://docs.travis-ci.com/user/sonarcloud/#Upcoming-improvements):
> Support for external pull requests.

**Pull request closed without merging**